### DISCUSSION TITLE
Keyboard problem ST7789 driver

### DISCUSSION THREAD

**Discussion by salarizadi** [Moddable-OpenSource/moddable#1176] [Q&A]
Hello.

Unfortunately, when I put the page in landscape mode, this problem occurs:

https://github.com/Moddable-OpenSource/moddable/assets/67143370/23623b07-4dac-48ad-bea6-d332dcf73f18

But when placed vertically, it works fine, where's the problem?

@phoddie

**Comment by phoddie** [comment]
@salarizadi – a couple questions:

- Which example app are you running?
- What are the dimensions of the display? For example 320x240.

**Reply by salarizadi** [reply]
- Keyboard example in the example folder
- Yes, size 320x240

**Reply by phoddie** [reply]
I just ran $MODDABLE/examples/piu/keyboard on a Moddable Two and in the simulator. In both cases, it works as expected. The touch registration and drawing is correct.

The ST7789 driver is just an alias for the ILI9341 driver used in Moddable Two, so they are identical apart from configuration. My guess is that the drawing problems are caused by an incorrect configuration in your display driver. 

The touch registration issue could be related to calibration. Did you have a chance to try running the XPT2046 calibration example?

**Reply by salarizadi** [reply]
I tried a lot to adjust it. It is interesting that it works without problems in horizontal mode, but in horizontal mode it only has this problem. The settings I made to calibrate:

https://github.com/Moddable-OpenSource/moddable/discussions/1175#discussioncomment-6462561

**Reply by salarizadi** [reply]
@phoddie Interesting, watch the clip below. When I removed the code (this.data["LABEL"].string = theString) it worked, why?

https://github.com/Moddable-OpenSource/moddable/assets/67143370/8825eecd-dd7e-4744-8ba0-0f12d5b874a4
